### PR TITLE
Intake and Sensors

### PR DESCRIPTION
**Pull request by Yumnah2**

### CODE REVIEW

File: src/main/java/frc/team2412/robot/commands/intake/AllInCommand.java [hunk 1]
```diff
@@ -34,6 +34,9 @@ public void end(boolean interrupted) {
 
 	@Override
 	public boolean isFinished() {
-		return intakeSubsystem.getFeederSensor();
+		if (!intakeSubsystem.getSensorOverride()) {
```
Comment (TAKBS2412) [inline src/main/java/frc/team2412/robot/commands/intake/AllInCommand.java]: This can be rewritten to use ||.

Comment (eddiert1239) [reply src/main/java/frc/team2412/robot/commands/intake/AllInCommand.java]: yeah you're right

File: src/main/java/frc/team2412/robot/commands/intake/AllInCommand.java [hunk 2]
```diff
@@ -34,6 +34,6 @@ public void end(boolean interrupted) {
 
 	@Override
 	public boolean isFinished() {
-		return intakeSubsystem.getFeederSensor();
+		return !intakeSubsystem.getSensorOverride() || intakeSubsystem.getFeederSensor();
```
Comment (TAKBS2412) [inline src/main/java/frc/team2412/robot/commands/intake/AllInCommand.java]: I think this is different from the older version of the logic, if sensor override is true but feeder sensor is false, the old version would return true but the new version returns false.

Comment (eddiert1239) [reply src/main/java/frc/team2412/robot/commands/intake/AllInCommand.java]: yeah you're right

File: src/main/java/frc/team2412/robot/commands/intake/AllInCommand.java [hunk 3]
```diff
@@ -20,7 +20,7 @@ public void initialize() {
 
 	@Override
 	public void execute() {
-		if (intakeSubsystem.getIndexSensor()) {
+		if (intakeSubsystem.getIndexSensor() && !intakeSubsystem.getSensorOverride()) {
```
Comment (bobtseattle) [inline src/main/java/frc/team2412/robot/commands/intake/AllInCommand.java]: To make the logic more clear, is it better to override the sensor here or in the getSensorOverride method?
Or with a parameter to getSensorOverride that overrides the sensor?

**Review by bobtseattle — COMMENTED** [review]
I am debating if there is a better way to override the sensor value so we don't have to implement extra logic in this and other commands(if ever needed)?

**Comment by eddiert1239** [comment]
> I am debating if there is a better way to override the sensor value so we don't have to implement extra logic in this and other commands(if ever needed)?

Both the getFeederSensor() and getIndexSensor() methods now return sensor value if not overriding, and false if we are overriding. This means no extra logic in commands.

**Review by bobtseattle — COMMENTED** [review]
LGTM.  Would be nice if the sensor method would return true to make it more intuitive, but we can go with the ! until we get confused.

File: src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java [hunk 4]
```diff
@@ -214,12 +223,16 @@ public void feederShoot() {
 	}
 
 	// sensor methods
-	public boolean getIndexSensor() {
-		return indexSensor.get();
+	public boolean indexSensorHasNote() {
+		return !indexSensor.get() && !getSensorOverride();
```
Comment (KangarooKoala) [inline src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:227]: Why is `indexSensor.get()` inverted now? (This also applies to `feederSensorHasNote()`)

**Review by KangarooKoala — COMMENTED** [review]
Looks pretty good to me, except the one question (and also the correct spelling is ingest, not injest).

File: src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java (hunk 4 above)
Comment (eddiert1239) [reply src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:227]: .get() returns true as a default, and if it senses something it returns false

**Review by TAKBS2412 — APPROVED** [review]

**Review by jamesdooley4 — APPROVED** [review]

**Pull request merged**